dpti/gdi.py

```diff
@@ -225,63 +239,69 @@ def make_dpdt (temp,
     if new_task :
         if verbose :
             print('# dpdt: do not find any matched record, run new task from %d ' % counter)
-        jdata = json.load(open('in.json', 'r'))        
+
+        with open('in.json', 'r') as j:
+            jdata = json.load(j)
         # make new task
-        work_path = os.path.join('database', 'task.%06d' % counter)
+        work_base = os.path.abspath(os.path.join('database', 'task.%06d' % counter))
+
         _make_tasks_onephase(temp, pres, 
-                             os.path.join(work_path, '0'),
+                             os.path.join(work_base, '0'),
                              jdata,
                              ens = jdata['phase_i'].get('ens', None),
                              conf_file = conf_0,
                              graph_file = 'graph.pb',
                              if_meam=if_meam,
                              meam_model=meam_model)
         _make_tasks_onephase(temp, pres, 
-                             os.path.join(work_path, '1'),
+                             os.path.join(work_base, '1'),
                              jdata, 
                              ens = jdata['phase_ii'].get('ens', None),
                              conf_file = conf_1,
                              graph_file = 'graph.pb',
                              if_meam=if_meam,
                              meam_model=meam_model)
         # submit new task
-        resources_dict = mdata['resources']
-        batch_dict = mdata['batch']
-        resources = Resources(**resources_dict)
-        batch = BatchObject(jdata=batch_dict)
+
+        # if workflow is None:
+        machine = Machine.load_from_machine_dict(mdata)
+        resources = machine.resources
+        batch = machine.batch
+
         command = 'lmp -i in.lammps'
-        # resources = mdata['resources']
-        # lmp_exec = mdata['command']
-        # command = lmp_exec + " -i in.lammps"
         forward_files = ['conf.lmp', 'in.lammps', 'graph.pb']
         if if_meam:
             meam_library_basename = os.path.basename(meam_model['library'])
             meam_potential_basename = os.path.basename(meam_model['potential'])
             forward_files.extend([meam_library_basename, meam_potential_basename])
         backward_files = ['log.lammps', 'out.lmp']
 
-        task1 = Task(
+        task0 = Task(
             command=command,
             task_work_path='0/',
             forward_files=forward_files,
             backward_files=backward_files
         )
-        task2 = Task(
+        task1 = Task(
             command=command,
             task_work_path='1/',
             forward_files=forward_files,
             backward_files=backward_files
         )
```

Comment:
I would prefer creating a list of `Task`s, so the implementation can be easily generalized to an arbitrary number of `Task`s